tools/cli/commands/clean.js

```diff
@@ -0,0 +1,607 @@
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Handle args for clean command.
+ *
+ * @param {argv}  argv - the arguments passed.
+ */
+export async function cleanCli( argv ) {
+	if ( argv.project ) {
+		await parseProj( argv );
+	} else {
+		await promptProj( argv );
+	}
+
+	if ( argv.include ) {
+		await parseToClean( argv );
+	} else {
+		await promptForClean( argv );
+	}
+
+	const allFiles = await collectAllFiles( argv.toClean );
+	const toCleanFiles = await collectCleanFiles( allFiles, argv.toClean );
+	const runConfirm = await confirmRemove( argv );
+
+	if ( ! runConfirm.confirm ) {
+		console.log( chalk.red( 'Cancelling jetpack clean.' ) );
+		return;
+	}
+
+	await cleanFiles( toCleanFiles, argv );
+	return;
+}
+
+/**
+ * Delete the files that we want.
+ *
+ * @param {Array} toCleanFiles - files that we want to clean.
+ * @param {object} argv - the arguments passed.
+ */
+async function cleanFiles( toCleanFiles, argv ) {
+	for ( const file of toCleanFiles ) {
+		fs.rm( file, { recursive: true, force: true }, ( err ) => {
+			console.log( 'Cleaning ', file );
+			if ( err ) {
+				// File deletion failed
+				console.error( err.message );
+				return;
+			}
+		} );
+	}
+
+	// Cleanup root and tools node_modules folder if that's what we're deleting.
+	if ( argv.scope === 'all' && argv.toClean.includes( 'node_modules' ) ) {
+		process.on( 'exit', async () => {
+			await runCommand( 'rm', [ '-rf', 'node_modules', 'tools/cli/node_modules' ] );
+		} );
+	}
+}
+/**
+ * Returns list of files that we want to delete.
+ *
+ * @param {Array} allFiles - a list of all possible deletable files.
+ * @param {Array} toClean - what kind of files we want to delete.
+ * @returns {Array} deleteQueue - files that we want to delete.
+ */
+async function collectCleanFiles( allFiles, toClean ) {
+	const deleteQueue = [];
+	for ( const file of toClean ) {
+		switch ( file ) {
+			case 'untracked':
+				deleteQueue.push( ...allFiles.untracked );
+				break;
+			case 'docker':
+				deleteQueue.push( ...allFiles.docker );
+				break;
+			case 'node_modules':
+				deleteQueue.push( ...allFiles.node_modules );
+				break;
+			case 'composer.lock':
+				deleteQueue.push( ...allFiles.composerLock );
+				break;
+			case 'vendor':
+				deleteQueue.push( ...allFiles.vendor );
+				break;
+			case 'other':
+				deleteQueue.push( ...allFiles.other );
+				break;
+		}
+	}
+
+	for ( const file of deleteQueue ) {
+		console.log( file );
+	}
+
+	return deleteQueue;
+}
+/**
+ * Gets list of files that could be deleted.
+ *
+ * @param {Array} toClean - files that we want to clean.
+ * @returns {object} allFiles.
+ */
+async function collectAllFiles( toClean ) {
+	const allFiles = {
+		untracked: [],
+		docker: [],
+		node_modules: [],
+		vendor: [],
+		composerLock: [],
+		other: [],
+		combined: [],
+	};
+
+	if ( toClean.includes( 'untracked' ) ) {
+		allFiles.untracked = child_process.execSync( 'git ls-files --exclude-standard --directory --other' );
```

Comment:
At some point you'll need to filter by `argv.project`. This seems like the obvious place since you can just include it in the command, like
```
allFiles.untracked = child_process.spawnSync( 'git', [ 'ls-files', '--exclude-standard', '--directory', '--other', argv.project ] );
```

Then similarly on line 177.